### PR TITLE
Fix cropping2D empty list #15325

### PR DESCRIPTION
**Pull request by arubiales**
This is the solution for the `Cropping2D` layer, following the same way we did in  #14970

@fchollet  @mattdangerw

### CODE REVIEW

File: keras/layers/convolutional.py [hunk 1]
```diff
@@ -3522,6 +3522,12 @@ def compute_output_shape(self, input_shape):
   def call(self, inputs):
     # pylint: disable=invalid-unary-operand-type
     if self.data_format == 'channels_first':
+      if (inputs.shape[2] is not None and inputs.shape[3] is not None and
```
Comment (fchollet) [inline keras/layers/convolutional.py:3525]: Presumably this should be

```
if ((inputs.shape[2] is not None and sum(self.cropping[0]) >= inputs.shape[2]) or
    (inputs.shape[3] is not None and sum(self.cropping[1]) >= inputs.shape[3]))):
```

File: keras/layers/convolutional.py [hunk 2]
```diff
@@ -3533,6 +3539,12 @@ def call(self, inputs):
       return inputs[:, :, self.cropping[0][0]:-self.cropping[0][1],
                     self.cropping[1][0]:-self.cropping[1][1]]
     else:
+      if (inputs.shape[1] is not None and inputs.shape[2] is not None and
```
Comment (fchollet) [inline keras/layers/convolutional.py:3542]: Likewise

**Review by fchollet — COMMENTED** [review]
Thanks for the PR

File: keras/layers/convolutional.py (hunk 2 above)
Comment (arubiales) [reply keras/layers/convolutional.py]: Did it both changes!

File: keras/layers/convolutional.py [hunk 3]
```diff
@@ -3514,6 +3514,11 @@ def compute_output_shape(self, input_shape):
   def call(self, inputs):
     # pylint: disable=invalid-unary-operand-type
     if self.data_format == 'channels_first':
+      if ((inputs.shape[2] is not None and sum(self.cropping[0]) >= inputs.shape[2]) or
+          (inputs.shape[3] is not None and sum(self.cropping[1]) >= inputs.shape[3])):
+          raise ValueError('cropping parameter of Cropping layer must be '
```
Comment (fchollet) [inline keras/layers/convolutional.py:3519]: "Argument `cropping`" (with backticks) 

"of Cropping layer" is not necessary, the traceback will show the origin

File: keras/layers/convolutional.py [hunk 4]
```diff
@@ -3514,6 +3514,11 @@ def compute_output_shape(self, input_shape):
...
+                        'greater than the input shape. Received: inputs.shape='
```
Comment (fchollet) [inline keras/layers/convolutional.py:3520]: Fix indent here and below

File: keras/layers/convolutional.py [hunk 5]
```diff
@@ -3525,6 +3530,11 @@ def call(self, inputs):
       return inputs[:, :, self.cropping[0][0]:-self.cropping[0][1],
                     self.cropping[1][0]:-self.cropping[1][1]]
     else:
+      if ((inputs.shape[1] is not None and sum(self.cropping[0]) >= inputs.shape[1]) or
+          (inputs.shape[2] is not None and sum(self.cropping[1]) >= inputs.shape[2])):
+          raise ValueError('cropping parameter of Cropping layer must be '
```
Comment (fchollet) [inline keras/layers/convolutional.py:3535]: Same comments here

**Review by fchollet — COMMENTED** [review]
Thanks for the update!

File: keras/layers/convolutional.py (hunk 5 above)
Comment (arubiales) [reply keras/layers/convolutional.py]: Did it! please check if it's correct.

**Review by fchollet — APPROVED** [review]
LGTM, thank you!

**Pull request merged**